experiments/experiment.go

```diff
@@ -64,21 +71,49 @@ func (ec *ExperimentController) reconcileTemplate(experiment *v1alpha1.Experimen
 	return true, nil
 }
 
+func (ec *ExperimentController) checkEnqueueExperimentDuringRun(experiment *v1alpha1.Experiment) {
```

Comment:
Just a suggestion. It feels like `experiment.PassedDurations`  has `checkEnqueueExperimentDuringRun` has the shared logic. To get rid of duplication `experiment.PassedDurations` might return `true/false` and remaining duration which can be used in `checkEnqueueExperimentDuringRun`. So `checkEnqueueExperimentDuringRun` would look like:

```
func (ec *ExperimentController) checkEnqueueExperimentDuringRun(experiment *v1alpha1.Experiment) {
  if passed, timeRemaining := experiment.PassedDurations(experiment); !passed {
    logCtx.Infof("Enqueueing Experiment in %s seconds", timeRemaining.String())
    ec.enqueueExperimentAfter(experiment, timeRemaining)
  }
}
```

Reply:
👍